lineairdb_field.h

```diff
@@ -0,0 +1,55 @@
+#include <string>
+#include "my_inttypes.h"
+
+/**
+ * @brief This class is responsible for the translation between
+ * MySQL Field value and LineairDB Field.
+ * @details
+ * LineairDB field consists of the following 3 information:
+ *  header1     header2
+ * [byteSize][valueLength][value]
+ * header info
+ * - byteSize: number of bytes of `valueLength`
+ *             always 1 byte, byteSize = UCHAR_MAX if valueLength = 0
+ * - valueLength: length of value
+ *                max 8 bytes
+ * value: MySQL value shown to users
+ *        max 4294967295 = sizeof(LONGBLOB) bytes
+ * Each row consists of multiple fields.
+ * First field stores null flags.
+ */
+class Lineairdb_Field
+{
+public:
+  std::string convert_numeric_to_bytes (const size_t num);
+  template <typename BYTE_TYPE>
+  size_t convert_bytes_to_numeric(const BYTE_TYPE* bytes, const size_t length);
+
+  /**
+   * @brief These methods are called for INSERT and UPDATE statements
+   */
+  std::string get_lineairdb_field();
+  template <typename BYTE_TYPE>
+  void set_lineairdb_field(const BYTE_TYPE* srcMysql, const size_t length);
+
+  /**
+   * @brief These methods are called for SELECT statements.
+   */
+  void make_mysql_table_row(const std::byte* const ldbRawData, const size_t length);
+  std::string get_null_flags();
+  std::string& get_column_of_row(size_t i);
+
+  Lineairdb_Field() {}
+  
+private:
+  static constexpr char noValue = 0xff;
```

Comment:
Is there any possibility that the same value is inputted as a user-input-value?

Reply:
`noValue` is only stored in header1 `byteSize`.
`byteSize` describes the length of header2 `valueLength`, which is max 8 bytes.
The value of `byteSize` logically makes sense only when it is between 1 to 8.
Therefore, `noValue` will never be an user-input-value.

I do not remember clearly why I used 0xff instead of 0 for `noValue`, but it may be related to 0 being treated as '\0' in `std::string`. It is safer to use 0xff, or any value greater than 8 (8 < noValue <= 0xff).

Reply:
I just realized that `valueLength` only needs to be 4 bytes long, as the maximum length of a MySQL column is only 4294967295 (`UINT_MAX`) bytes long.
I thought that 4294967295 = `ULONG_MAX`, and that was the reason why I had `valueLength` to be max 8 bytes.
Sorry for the mistake and confusion.

Reply:
Okay.

I recommend you use assertion for each input value to care about the magic number `0xff`.